src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInMetricsListener.cs

```diff
@@ -50,6 +54,12 @@ public override void OnEventWritten(string name, object payload)
                     return;
                 }
 
+                if (this.options.Filter?.Invoke(context) == false)
```

Comment:
We should wrap the calls into user code with try/catch. Here's a couple examples you can follow:

* https://github.com/open-telemetry/opentelemetry-dotnet/blob/39283c7327435b1d3ca9702387363030226b45a6/src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInListener.cs#L174-L190

* https://github.com/open-telemetry/opentelemetry-dotnet/blob/39283c7327435b1d3ca9702387363030226b45a6/src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInListener.cs#L223-L230

Reply:
Added as suggested